### ISSUE TITLE
Can't get this to compile - no matching function for call to 'SDClass::begin()'

### ISSUE THREAD

**Issue by panicacid** [eggsactly/arduino-wardriving#2]
Not sure what libraries you were using for this build but I can't get it to build in Arduino constantly getting 

no matching function for call to 'SDClass::begin()'

Don't suppose you can remember / know what libraries you're using for everything?

**Comment by eggsactly** [comment]
Hello @panicacid , sorry about the delay. I've been without Internet for two months. 

I built this project for Adafruit's 8266 Feather. Are you using the *Adafruit Feather HUZZAH ESP8266* board package?

**Comment by panicacid** [comment]
> Hello @panicacid , sorry about the delay. I've been without Internet for two months.
> 
> I built this project for Adafruit's 8266 Feather. Are you using the _Adafruit Feather HUZZAH ESP8266_ board package?

Indeed. Using everything you used, got everything cut and soldered but can't compile and flash. 

It looks like some libraries have been updated and deprecated since, far as I can tell you used the adafruit library for the SD logging but it seems adafruit deprecated that in favour of the esp8266 libraries sd code.

**Comment by eggsactly** [comment]
What happens when you try to build the listfiles example in the SD section of examples for any board with the HUZZAH ESP8266 board package?

**Comment by eggsactly** [comment]
@panicacid let me know how you're doing. If I hear nothing, I'll close this issue on February 24th, 2022.

**Comment by panicacid** [comment]
Good Morning Eggsactly, apologies for going quiet on this one, other projects and life got in the way. However! I'm back at it.

So I've installed the latest ESP8266 board package, and additional libraries that were released just before you published this in 2019

RTClib-1.2.0
Adafruit_FeatherOLED-1.0.0
TinyGPSPlus-1.0.2

I'm also running 

Adafruit GFX Library 1.10.14
Adafruit SSD1306 2.4.7
OneWire 2.3.7

However, upon compiling I'm got:

C:\Users\*SNIP*\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\3.0.2\libraries\SD\src/SD.h:35:13: note:   candidate expects 2 arguments, 0 provided
Multiple libraries were found for "SD.h"
 Used: C:\Users\*SNIP*\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\3.0.2\libraries\SD
 Not used: C:\Program Files (x86)\Arduino\libraries\SD
exit status 1
no matching function for call to 'SDClass::begin()'

So I removed version 3.0.2 of the esp8266 board package and installed 2.5.0 and it compiled!

**Comment by eggsactly** [comment]
Very interesting. I'll have to setup the Arduino IDE and see if I can get the same results. Perhaps the SD API has changed and I'll need to update this code.
